ocis-pkg/runner/grouprunner.go

```diff
@@ -0,0 +1,130 @@
+package runner
+
+import (
+	"context"
+)
+
+// GroupRunner represent a group of tasks that need to run together.
+// The expectation is that all the tasks will run at the same time, and when
+// one of them stops, the rest will also stop.
+//
+// The GroupRunner is intended to be used to run multiple services, which are
+// more or less independent from eachother, but at the same time it doesn't
+// make sense to have any of them stopped while the rest are running.
+// Basically, either all of them run, or none of them.
+// For example, you can have a GRPC and HTTP servers running, each of them
+// providing a piece of functionality, however, if any of them fails, the
+// feature provided by them would be incomplete or broken.
+//
+// Note that, as services, the task aren't expected to stop by default.
+// This means that, if a task finishes naturally, the rest of the task will
+// asked to stop as well.
+type GroupRunner struct {
+	runners []*Runner
+}
+
+// NewGroupRunner will create a GroupRunner
+func NewGroupRunner() *GroupRunner {
+	return &GroupRunner{
+		runners: []*Runner{},
+	}
+}
+
+// Add will add a runner to the group.
+//
+// It's mandatory that each runner in the group has an unique id, otherwise
+// there will be issues
```

Comment:
I'm not so sure about this one...
On the one hand, if we enforce it, we'll have to return an error on failure, which means we'll have to check for the error when we're creating the runner.
On the other hand, the unique id restriction is solved during coding. I'd expect up to 4-5 runners in the same group, so it should be easy to keep their ids unique. There is a risk involved, but I think the code is easier to follow.

Reply:
Mmh. I get your point, but I would prefer to have it enforced. There is only this one comment that tells users to use unique ids and there is no obvious panic or error that would tell the user what is wrong. If you don't want to care about the error we can log and `os.Exit` instead of returning an error

Reply:
I'll throw a panic if there are duplicate ids. It seems reasonable